olp-cpp-sdk-core/include/olp/core/thread/ExecutionContext.h

```diff
@@ -0,0 +1,86 @@
+/*
+ * Copyright (C) 2022 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#pragma once
+
+#include <utility>
+
+#include <olp/core/client/ApiError.h>
+#include <olp/core/client/CancellationContext.h>
+
+namespace olp {
+namespace thread {
+
+using FailedCallback = std::function<void(client::ApiError)>;
+using CancelFuncType = std::function<void()>;
+
+/**
+ * * @brief The `ExecutionContext` class is used to handle cancellation, final,
+ * and failed mechanisms
+ */
+class CORE_API ExecutionContext {
+  using ExecuteFuncType = std::function<client::CancellationToken()>;
+
+ public:
+  /// Default contructor, initializes ExecutionContextImpl instance.
+  ExecutionContext();
+
+  /**
+   * @brief Gets the error object
+   * @return The `ApiError` instance.
+   */
+  const client::ApiError GetError() const;
+
+  /**
+   * @brief Sets an error which will be returned in Finally method of
+   * Continuation.
+   *
+   * @param error The `ApiError` instance.
+   */
+  void SetError(client::ApiError error);
+
+  /**
+   * @brief Shows whether or not an error was set.
+   * @return True if an error was set.
+   */
+  bool Failed() const;
+
+  /// @copydoc ContinuationImpl::Cancelled() const
```

Comment:
There is no such thing in the current code-base. Check other methods below as well

Reply:
done